File: advent-of-code/advent_of_code_2016/day_3/part_1.js
function fileRead(filename) {
  var contents = fs.readFileSync(filename);
  return contents.toString();
}

var fs = require('fs');
var input = fileRead('input.txt').split('\n');

var possible_triangles = input.map((str) => str.match(/\w+/g));

possible_triangles.pop(); // for some reason, there is an unwanted null value at the end

let count = 0

possible_triangles.forEach(function(arr) {
  sides = arr.map((str) => Number(str)).sort((a, b) => a - b);
  if (sides[0] + sides[1] > sides[2]) count += 1;
});

console.log(count);

File: advent-of-code/advent_of_code_2016/day_3/part_1.rb
input = File.read('input.txt').split("\n")
possible_triangles = input.map { |sides| sides.scan(/\w+/).map(&:to_i).sort }

p possible_triangles.size

p possible_triangles.select { |sides| sides[0] + sides[1] > sides[2] }.size
